core/Command/User/LastSeen.php

```diff
@@ -31,55 +33,75 @@
 use Symfony\Component\Console\Input\InputArgument;
 use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
 
-class LastSeen extends Base {
-	public function __construct(
-		protected IUserManager $userManager,
-	) {
-		parent::__construct();
-	}
+class LastSeen extends Base
+{
+    public function __construct(
+        protected IUserManager $userManager,
+    ) {
+        parent::__construct();
+    }
 
-	protected function configure() {
-		$this
-			->setName('user:lastseen')
-			->setDescription('shows when the user was logged in last time')
-			->addArgument(
-				'uid',
-				InputArgument::REQUIRED,
-				'the username'
-			);
-	}
+    protected function configure()
+    {
+        $this
+            ->setName('user:lastseen')
+            ->setDescription('shows when the user was logged in last time')
+            ->addArgument(
+                'uid',
+                InputArgument::OPTIONAL,
+                'the username'
+            )
+            ->addOption(
+                'all',
+                null,
+                InputOption::VALUE_NONE,
+                'shows a list of when all users were last logged in'
+            )
+        ;
+    }
 
-	protected function execute(InputInterface $input, OutputInterface $output): int {
-		$user = $this->userManager->get($input->getArgument('uid'));
-		if (is_null($user)) {
-			$output->writeln('<error>User does not exist</error>');
-			return 1;
-		}
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $singleUserId = $input->getArgument('uid');
+        if ($singleUserId) {
+            if (is_null($this->userManager->get($singleUserId))) {
+                $output->writeln('<error>User does not exist</error>');
+                return 1;
+            }
+            $users = $this->userManager->search($singleUserId);
+        } elseif ($input->getOption('all')) {
+            $users = $this->userManager->search('');
+        } else {
+            $output->writeln("<error>Please specify a username, or \"--all\" to list all</error>");
+            return 1;
+        }
 
-		$lastLogin = $user->getLastLogin();
-		if ($lastLogin === 0) {
-			$output->writeln('User ' . $user->getUID() .
-				' has never logged in, yet.');
-		} else {
-			$date = new \DateTime();
-			$date->setTimestamp($lastLogin);
-			$output->writeln($user->getUID() .
-				'`s last login: ' . $date->format('d.m.Y H:i'));
-		}
-		return 0;
-	}
+        foreach($users as $user) {
+            $lastLogin = $user->getLastLogin();
+            if ($lastLogin === 0) {
+                $output->writeln($user->getUID() . ' has never logged in.');
+            } else {
+                $date = new \DateTime();
+                $date->setTimestamp($lastLogin);
+                $output->writeln($user->getUID() . "'s last login: " . $date->format('Y-m-d H:i'));
```

Comment:
I think the next feature request will be to be able to sort by last login, so maybe we should either sort, or do something like that so that it can be piped into sort:
```suggestion
                $output->writeln($date->format('Y-m-d H:i').' is the last login from '.$user->getUID());
```